src/WorkflowCore/Services/WorkflowController.cs

```diff
@@ -85,7 +83,11 @@ public async Task<string> StartWorkflow<TData>(string workflowId, int? version,
 
             wf.ExecutionPointers.Add(_pointerFactory.BuildGenesisPointer(def));
 
-            await _middlewareRunner.RunPreMiddleware(wf, def);
+            using (var scope = _serviceProvider.CreateScope())
```

Comment:
Should we use `IScopeProvider` instead?

Reply:
I think it wouldn't be possible here because the scope provider relies on passing an `IStepExecutionContext`